### PR TITLE
Add `IPermissionCommand`

### PR DESCRIPTION
**Pull request by Vladislav-CS**
Not tested

### CODE REVIEW

File: Exiled.Events/Patches/Generic/ProcessQuery.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    /// Implements <see cref="IPermissionCommand.Permission" />.
+    /// </summary>
+    [HarmonyPatch(typeof(CommandProcessor), nameof(CommandProcessor.ProcessQuery))]
+    internal class ProcessQuery
+    {
+        private static bool CheckPermissions(ICommand command, CommandSender sender)
+        {
+            if (command is IPermissionCommand permissionCommand && !sender.CheckPermission(permissionCommand.Permission))
+            {
+                sender.RaReply($"{command.Command}#You do not have permissions to use this command", false, true, string.Empty);
```
Comment (VALERA771) [inline Exiled.Events/Patches/Generic/ProcessQuery.cs:33]: Maybe edit to show which permission user need to use this command?

File: Exiled.Events/Patches/Generic/ProcessQuery.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+                // bool hasPermissions = CheckPermissions(command, sender)
+                new(OpCodes.Call, Method(typeof(ProcessQuery), nameof(CheckPermissions))),
+                new(OpCodes.Stloc_S, hasPermissions),
+
+                // hasPermissions
+                new(OpCodes.Ldloc_S, hasPermissions),
+
+                // if (!hasPermissions) return;
+                new(OpCodes.Brfalse_S, returnLabel),
```
Comment (VALERA771) [inline Exiled.Events/Patches/Generic/ProcessQuery.cs:67]:
```suggestion
                new(OpCodes.Brfalse_S, returnLabel),
```

File: Exiled.Events/Patches/Generic/ProcessQuery.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+            newInstructions.InsertRange(index + offset, new CodeInstruction[]
+            {
+                // command
+                new(OpCodes.Ldloc_1),
+
+                // sender
+                new(OpCodes.Ldarg_1),
+
+                // bool hasPermissions = CheckPermissions(command, sender)
```
Comment (VALERA771) [inline Exiled.Events/Patches/Generic/ProcessQuery.cs:59]:
```suggestion
                // if (!CheckPermissions(command, sender))
                //    return;
```

File: Exiled.Events/Patches/Generic/ProcessQuery.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            }
+
+            return true;
+        }
+
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            LocalBuilder hasPermissions = generator.DeclareLocal(typeof(bool));
```
Comment (VALERA771) [inline Exiled.Events/Patches/Generic/ProcessQuery.cs:44]:
```suggestion
```

**Review by VALERA771 — COMMENTED** [review]
2 suggestions to optimize IL code and one for better implementation

**Comment by NaoUnderscore** [comment]
@Vladislav-CS Run some tests if you can ❤️

**Review by ALEXWARELLC — APPROVED** [review]
Tested - Working.

**Pull request merged**